src/components/CheckBox/CheckBox.tsx

```diff
@@ -50,7 +52,7 @@ const CheckBox: React.FC<Props> = ({
           type="checkbox"
           onChange={handleInputChange}
           disabled={disabled}
-          checked={isChecked}
+          checked={checked}
```

Comment:
Since we are using checked prop, I think we don't need the useState anymore.

Reply:
it doesnt hurt to have controlled and uncontrolled version